src/utils/fileMatchesConfigGlob.ts

```diff
@@ -21,5 +21,5 @@ export default function fileMatchesConfigGlob(
 function removeFileExtension(filePath: NormalizedPath): NormalizedPath {
     // Special case for .d.ts files
     let extension = filePath.endsWith('.d.ts') ? '.d.ts' : path.extname(filePath);
-    return <NormalizedPath>filePath.slice(0, -extension.length);
+    return <NormalizedPath>filePath.slice(0, filePath.length - extension.length);
```

Comment:
There was a bug here with extension-less paths (since -0 is just 0 as far as slice is concerend).

I fixed the bug was calling this function with extension-less paths, but left this fix in.